src/wish.cpp

```diff
@@ -476,36 +485,42 @@ void debug_menu::wishitem( player *p, int x, int y, int z )
         wmenu.query();
         if( wmenu.ret >= 0 ) {
             item granted( opts[wmenu.ret] );
+            if( dynamic_cast<wish_item_callback *>( wmenu.callback )->incontainer ) {
```

Comment:
Well, I actually moved these code here from a latter place. I'll see if I can improve it.